ex03.exs

```diff
@@ -53,9 +53,32 @@ defmodule Ex03 do
   def pmap(collection, process_count, function) do
     # your code goes here
     # I'm hoping to see a simple pipeline as the body of this function...
+    collection
+    |> Ex03.splitbyprocess(process_count)
+    |> Ex03.mapfunctions(function)
+    |> List.flatten()
   end
 
-  # and here...
+  def splitbyprocess(collection, process_num) do
+    Enum.chunk_every(collection, div(Enum.count(collection),process_num))
```

Comment:
The `div` returns the wrong answer: `div(10,4)` returns 2, which means you'd get 5 processes. It should be

    div(Enum.count(collection) + process_count - 1, process_count)

# -2